plugins/camera/camera_impl.cpp

```diff
@@ -0,0 +1,154 @@
+#include "camera_impl.h"
+#include "device.h"
+#include "global_include.h"
+#include "mavlink_include.h"
+#include <functional>
+
+namespace dronecore {
+
+CameraImpl::CameraImpl(Device &device) :
+    PluginImplBase(device)
+{
+    _parent.register_plugin(this);
+}
+
+CameraImpl::~CameraImpl()
+{
+    _parent.unregister_plugin(this);
+}
+
+void CameraImpl::init() {}
+
+void CameraImpl::deinit() {}
+
+void CameraImpl::enable() {}
+
+void CameraImpl::disable() {}
+
+Camera::Result CameraImpl::take_photo()
+{
+    return camera_result_from_command_result(
+               _parent.send_command_with_ack(
+                   MAV_CMD_IMAGE_START_CAPTURE,
+                   MavlinkCommands::Params {0.0f, // all camera IDs
```

Comment:
What is the difference between Camera ID and Camera Component ID ?

Reply:
Camera ID should actually be a thing of the past. I removed the comments.